### PR TITLE
dockerfile moved to debian

### PR DESCRIPTION
**Pull request by czervenka**

### CODE REVIEW

File: src/karmen_backend/Dockerfile [hunk 1]
```diff
@@ -1,41 +1,25 @@
...
 COPY Pipfile* ./
-RUN jq -r '.default | to_entries[] | .key + .value.version' \
-    Pipfile.lock > requirements.txt && pip install -r requirements.txt
-
-
-FROM python:3.7-alpine as runner
-RUN echo "http://dl-4.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
+RUN set -e && pipenv lock -r > requirements.pip && pip install -r requirements.pip
+# build deps
+RUN apt-get remove --purge --yes gcc make postgresql-client libpq-dev libffi-dev python3-dev libxml2-dev libxslt-dev
```
Comment (xaralis) [inline src/karmen_backend/Dockerfile:20]: I'm pretty sure that the app will crash in runtime without `libxml2-dev` and `libxslt-dev`.

File: src/karmen_backend/Dockerfile [hunk 2]
```diff
@@ -1,41 +1,25 @@
...
 
+RUN set -e && pip install --upgrade pip pipenv uwsgi supervisor
 # Install from lockfile
+# there is a bug in current pipenv version when using --system flag
 COPY Pipfile* ./
-RUN jq -r '.default | to_entries[] | .key + .value.version' \
-    Pipfile.lock > requirements.txt && pip install -r requirements.txt
-
-
-FROM python:3.7-alpine as runner
```
Comment (xaralis) [inline src/karmen_backend/Dockerfile:19]: Removing build docker stage goes directly against future docker layer caching as the resulting images will be a lot bigger in size.

File: src/karmen_backend/Dockerfile (hunk 1 above)
Comment (xaralis) [reply src/karmen_backend/Dockerfile:20]: Also, deleting this won't actually help as far as I can tell (but I can be wrong on this). The layer is already there previously.

File: src/karmen_backend/Dockerfile (hunk 2 above)
Comment (czervenka) [reply src/karmen_backend/Dockerfile:19]: It makes the resulting image slightly bigger. The caching is per `RUN` statement not per image. The caching should be the same.

File: src/karmen_backend/Dockerfile (hunk 1 above)
Comment (czervenka) [reply src/karmen_backend/Dockerfile:20]: ok, I thought that it is required during build time only

Comment (czervenka) [reply src/karmen_backend/Dockerfile]: I put it back in [7535c8](https://github.com/fragaria/karmen/pull/213/commits/7535c8aee112e5b962f847f6ea5e9bdb86a40638)

File: src/karmen_backend/Dockerfile (hunk 2 above)
Comment (xaralis) [reply src/karmen_backend/Dockerfile:19]: Yes, but the image to be downloaded to load the cached layers from will grow in size.

Comment (czervenka) [reply src/karmen_backend/Dockerfile:19]: Regarding cache size, it should be smaller if we remove the intermediate build server from dockerfile. The commands required to build the application will be cached in both cases. The difference is that in the later case (no intermediate build server) all layers will be shared with the final server (resulting in smaller cache but bigger final image). .... In theory. We need to verify it for sure.

**Comment by xaralis** [comment]
All that matters is the size that will be downloaded by image end users.
They will download all the layers of the final image.

Which is always smaller with separate build stage. That's actually the sole
purpose of having one: to avoid adding layers with build deps that are not
needed for running the app, just for building it.

**Comment by xaralis** [comment]
I didn't realise you only talk about CI. In this case, it's of course
better to have just a single image without the build stage. But we also
need to care about the final result for end users. This is only relevant
for final release images though.

Maybe we can have two docker files: one for dev and running the tests and
another one for final production image.

**Comment by czervenka** [comment]
this pull request stopped to build on travis for some reason. moved to [new pull request](https://github.com/fragaria/karmen/pull/224)

**Pull request closed without merging**